tools/toitlsp/lsp/compiler/file_server_pipe.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2021 Toitware ApS.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; version
+// 2.1 only.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
+package compiler
+
+import (
+	"io"
+	"sync"
+
+	"go.uber.org/zap"
+)
+
+type PipeFileServer struct {
+	l           sync.Mutex
```

Comment:
What does this lock cover?, if its `r` and `w` you should use the lock in `Run` and `IsReady` as well?

Reply:
done.